src/uiprotect/api.py

```
@@ -4580,13 +4605,21 @@ async def update_public(self) -> PublicBootstrap:
             if camera.rtsps_streams is not None
         }
 
+        # Seed the rate limiter before the fan-out: a single cheap fetch primes
+        # the header-driven budget (and the token bucket) so the concurrent
+        # burst below is paced from its first request instead of steering
+        # mid-flight. The NVR fetch doubles as that seed.
+        try:
+            pb.nvr = await self.get_nvr_public()
+        except BaseException as nvr_err:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `BaseException`

(BLE001)